main.go

```diff
@@ -173,58 +175,74 @@ func ndkVersion(ndkPath string) string {
 	return ""
 }
 
-func currentNDKHome() string {
-	if v := os.Getenv(androidNDKHome); v != "" {
-		return v
+// targetNDKPath picks the correct path where the requested NDK version should be installed,
+// as well as returning if the path needs to be cleaned up before install.
+// There are many (deprecated) ways to install NDK, the logic is based on the following:
+// https://github.com/android/ndk-samples/wiki/Configure-NDK-Path
+// https://developer.android.com/tools/variables
+func targetNDKPath(envRepo env.Repository, requestedNDKVersion string) (string, bool) {
+	if v := envRepo.Get(androidNDKHome); v != "" {
+		// $ANDROID_NDK_HOME is old and AGP no longer takes it into account,
+		// but it's an explicit path, so use it if it's set on the system.
+		// And because we don't know if it's `ndk-bundle` or a specific side-by-side version,
+		// we return true for cleanup.
+		log.Warnf("$%s is set to %s", androidNDKHome, v)
+		log.Warnf("This variable is deprecated and modern Android Gradle Plugin versions no longer take it into account.")
+		return v, true
 	}
-	if v := os.Getenv("ANDROID_HOME"); v != "" {
-		// $ANDROID_HOME is deprecated
-		return filepath.Join(v, "ndk-bundle")
+	if androidHome := envRepo.Get("ANDROID_HOME"); androidHome != "" {
+		// The most modern way is to install NDK versions side-by-side at $ANDROID_HOME/ndk/version
+		// This is what `sdkmanager` does when installing a specific version (`sdkmanager "ndk;26.3.11579264"`).
+		ndkPath := filepath.Join(androidHome, "ndk", requestedNDKVersion)
```

Comment:
This will only work if specifying the exact version number right? Dunno if it is a problem